src/lib/list/list.ts

```diff
@@ -50,6 +63,347 @@ export class MdList {
   set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
 }
 
+
+
+export interface MdSelectionListOptionEvent {
+  option: MdListOption;
+}
+
+/**
+ * Component for list-options of selection-list. Each list-option can automatically
+ * generate a checkbox and can put current item into the selectionModel of selection-list
+ * if the current item is checked.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'md-list-option',
+  host: {
+    'role': 'option',
+    'class': 'mat-list-item',
+    '(focus)': '_handleFocus()',
+    '(blur)': '_handleBlur()',
+    '(click)': 'toggle()',
+    'tabindex': '-1',
+    '[attr.aria-selected]': 'selected.toString()',
+    '[attr.aria-disabled]': 'disabled.toString()',
+  },
+  templateUrl: 'list-option.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class MdListOption implements AfterContentInit, OnDestroy, Focusable {
+  private _lineSetter: MdLineSetter;
+  private _disableRipple: boolean = false;
+  private _isSelectionList: boolean = false;
```

Comment:
You don't need `_isSelectionList` since these options can only appear in a selection list.